packages/dht/src/connection/websocket/WebsocketConnector.ts

```diff
@@ -192,8 +192,16 @@ export class WebsocketConnector {
                 version: LOCAL_PROTOCOL_VERSION
             }
         }
-        for (const reattempt of range(ENTRY_POINT_CONNECTION_ATTEMPTS)) {
-            const entryPoint = sample(this.config.entrypoints)!
+        const entrypoints = new Map<DhtAddress, PeerDescriptor>()
+        for (const entrypoint of this.config.entrypoints) {
+            entrypoints.set(getNodeIdFromPeerDescriptor(entrypoint), entrypoint)
+        }
+        const exclude: DhtAddress[] = []
+        const getExcludedIds = () => Array.from(entrypoints.keys()).filter((key) => !exclude.includes(key))
+        while (getExcludedIds().length > 0) {
+            const excludedIds = getExcludedIds()
+            const entryPointId = sample(excludedIds)!
+            const entryPoint = entrypoints.get(entryPointId)!
```

Comment:
Could use lodash/shuffle here to simplify?
```suggestion
        for (const entryPoint of shuffle(this.config.entrypoints)) {
```

Reply:
I'd prefer to shuffle first, always pick the first index and then remove it in a while loop to make excluding possible